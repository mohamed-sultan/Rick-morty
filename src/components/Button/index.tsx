import { styled } from "@mui/material/styles";
import Button from "@mui/material/Button";
import { ICustomButtonProps } from "../../interface";
const MaterialButton = styled(Button)(({ theme }) => ({
  background: theme.palette.primary?.black,
  color: theme.palette.primary?.white,
  cursor: "pointer",
  marginRight: "1rem",
  fontSize: "1.1rem",
  border: "unset",
  borderRadius: "10px",
  padding: " 0.3rem 0.9rem",
  transition: "all 0.5s ease",
  textTransform: "lowercase",
  "&:hover": {
    transform: "scale(1.1)",
    background: theme.palette.primary?.black,
    textDecoration: "none",
  },
}));

export default function CustomizedButton({ text }: ICustomButtonProps) {
  return (
    <MaterialButton variant="contained" disableRipple>
      {text}
    </MaterialButton>
  );
}
