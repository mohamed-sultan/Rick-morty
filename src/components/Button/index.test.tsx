import * as React from "react";
import TestRenderer from "react-test-renderer";
import Button from "./index";

describe("Button snapshot testing", () => {
  it("renders correctly", () => {
    const tree = TestRenderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
