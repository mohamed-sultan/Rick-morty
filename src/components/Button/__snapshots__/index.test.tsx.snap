// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot testing renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-qt782h-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
/>
`;
